### PR TITLE
`meltTokens` signs P2PK proofs

### PR DESCRIPTION
**Pull request by lollerfirst**
# Fixes: #164 

## Description
added option to sign P2PK proofs if a private key is provided

## PR Tasks

- [x] Open PR
- [x] run `npm run test` --> no failing unit tests
- [x] run `npm run format`

### CODE REVIEW

**Comment by lollerfirst** [comment]
@gudnuf Let me know if this helps...

**Review by gudnuf — APPROVED** [review]
ack! exactly what I need, thanks!

**Comment by lollerfirst** [comment]
@Egge21M can you bring this in?

File: src/CashuWallet.ts [hunk 1]
```diff
@@ -440,6 +442,19 @@ class CashuWallet {
...
 		);
+		if (options?.privkey != undefined) {
+			proofsToSend = getSignedProofs(
+				proofsToSend.map((p) => {
+					return {
+						amount: p.amount,
+						C: pointFromHex(p.C),
+						id: p.id,
+						secret: new TextEncoder().encode(p.secret)
+					};
```
Comment (callebtc) [inline src/CashuWallet.ts:453]: Could we refactor this into `getSignedProofs`?

Comment (callebtc) [reply src/CashuWallet.ts:453]: Just saw it's in cashu/crypto and the `Proof` type has a different definition:

`@cashu/cashu-ts`

```ts
 */
export type Proof = {
	id: string;
	amount: number;
	secret: string;
	C: string;
};
```

vs `@cashu/crypto`:

```ts
export type Proof = {
	C: ProjPointType<bigint>;
	secret: Uint8Array;
	amount: number;
	id: string;
	witness?: Witness;
};
```

☠️

Comment (Egge21M) [reply src/CashuWallet.ts:453]: Yes, thats not great. Unrelated to this PR, but we should add a utility that hides this back-and-forth between the two formats in the near future

**Comment by Egge21M** [comment]
@lollerfirst Thank you so much for taking this on. This does look good to me, but this branch diverges from the current development branch and a merge would add some duplicate commits under old hashes. Could you please reset this branch to 4ae5562496f221271e3f68025c325807d709121d and rebase it?

**Comment by lollerfirst** [comment]
@Egge21M Is this good?

**Review by Egge21M — APPROVED** [review]

**Pull request merged**